tests/view/test_slice.cpp

```diff
@@ -0,0 +1,119 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/view/extents.hpp"
+#include "kokkos-utils/view/slice.hpp"
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * Get slices of a @c Kokkos::View
+ * -------------------------------
+ *
+ * This group of tests check the behavior of our helpers related to @c Kokkos::View that can be
+ * found in @ref slice.hpp.
+ */
+
+namespace Kokkos::utils::tests::view
+{
+
+#define CHECK_THAT(__slice__, __rank__, __extents__)       \
+    {                                                      \
+        const auto tmp = __slice__;                        \
+        ASSERT_EQ(tmp.rank(), __rank__);                   \
+        ASSERT_EQ(utils::view::extents(tmp), __extents__); \
```

Comment:
Same for those :) May be worth trying with `static_assert`.

Reply:
I'm also wondering how it works with the dynamic extents. May be interesting to see whether the `static_assert` works with dynamic extents in particular.

Reply:
I won't work, not doing what you suggest.